packages/hierarchy-builder/src/test/internal/FilteringHierarchyLevelDefinitionsFactory.test.ts

```diff
@@ -0,0 +1,608 @@
+
+    function createTestNode(): HierarchyNode {
+      return {
+        key: "test",
+        label: "test node",
+        children: [],
+      };
+    }
+  });
+
+  describe("postProcessNode", () => {
+    it("returns given node when source factory has no post-processor", () => {
+      const node = createTestNode();
+      const filteringFactory = createFilteringHierarchyLevelsFactory();
+      const result = filteringFactory.postProcessNode(node);
+      expect(result).to.eq(node);
+    });
+
+    it("returns node post-processed by source factory", () => {
+      const inputNode = createTestNode();
+      const sourceFactoryNode = createTestNode();
+      const sourceFactory = {
+        postProcessNode: sinon.stub().returns(sourceFactoryNode),
+      } as unknown as IHierarchyLevelDefinitionsFactory;
+      const filteringFactory = createFilteringHierarchyLevelsFactory({
+        sourceFactory,
+      });
+      const result = filteringFactory.postProcessNode(inputNode);
+      expect(sourceFactory.postProcessNode).to.be.calledOnceWithExactly(inputNode);
+      expect(result).to.eq(sourceFactoryNode);
+    });
+
+    it("doesn't set auto-expand on class grouping nodes if none of the children have filtered children paths", () => {
+      const inputNode = createClassGroupingNode();
+      const filteringFactory = createFilteringHierarchyLevelsFactory();
+      const result = filteringFactory.postProcessNode(inputNode);
+      expect(result.autoExpand).to.be.undefined;
+    });
+
+    it("sets auto-expand on class grouping nodes if any child has filtered children paths", () => {
+      const inputNode: HierarchyNode = {
+        ...createClassGroupingNode(),
+        children: [
+          {
+            ...createTestNode(),
+            filteredChildrenIdentifierPaths: [],
+          } as FilteredHierarchyNode,
+        ],
+      };
+      const filteringFactory = createFilteringHierarchyLevelsFactory();
+      const result = filteringFactory.postProcessNode(inputNode);
+      expect(result.autoExpand).to.be.true;
+    });
+
+    function createTestNode(): HierarchyNode {
+      return {
+        key: "test",
+        label: "test node",
+        children: [],
+      };
+    }
+
+    function createClassGroupingNode(): HierarchyNode {
+      return {
+        ...createTestNode(),
+        key: {
+          type: "class-grouping",
+          class: { id: "0x1", label: "class label", name: "class name" },
+        },
+      };
+    }
+  });
+
+  describe("defineHierarchyLevel", () => {
+    const metadataProvider = {} as unknown as IMetadataProvider;
+    let stubClass: TStubClassFunc;
+    beforeEach(() => {
+      stubClass = createGetClassStub(metadataProvider).stubClass;
+    });
+
+    it("returns source definitions when filtered instance paths list is empty", async () => {
+      const sourceDefinitions: HierarchyLevelDefinition = [];
+      const sourceFactory = {
+        defineHierarchyLevel: sinon.stub().resolves(sourceDefinitions),
```

Comment:
Looks like `sinon` is not necessary in most of these test and this could be simplified to `defineHierarchyLevel: async () => sourceDefinitions`?

Reply:
agree, will clean this up

Reply:
done